azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/impl/MessagingFactory.java

```diff
@@ -543,15 +548,42 @@ public void run() {
                     return;
                 }
 
-                this.rctr.free();
-
                 if (getIsClosingOrClosed() && !closeTask.isDone()) {
+                    this.rctr.free();
+
                     closeTask.complete(null);
 
-                    if (closeTimer != null)
+                    if (closeTimer != null) {
                         closeTimer.cancel(false);
+                    }
+                } else {
+                    new ScheduledThreadPoolExecutor(1).schedule(new Runnable() {
```

Comment:
>new ScheduledThreadPoolExecutor(1) [](start = 20, length = 34)

what is the need for this?

Reply:
The question is that, original code is:
`reactor.free()`
This PR effectively changes this line to 
`ScheduledThreadPoolExecutor(1).schedule(new …..{ reactor.free()  });`

How does this help?
reactor.free is run regardless -  in both the cases!

Also, current eventhub client sdk - takes in an ExecutorService and doesn't spin off any threads in the SDk layer. Please think about similar strategy for this new ScheduledThreadPool.